File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;
import java.util.List;
import java.util.Map;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;

import entity.PageResult;
import entity.Result;

@RestController
@RequestMapping("brand")
public class BrandController {

	@Reference
	private BrandService brandService;

	@RequestMapping("findAll")
	public List<TbBrand> findAll() {
		return brandService.findAll();
	}

	@RequestMapping("findPage")
	public PageResult findPage(@RequestBody TbBrand tbBrand,int page, int pageSize) {
		PageResult pageResult = brandService.findPage(tbBrand,page, pageSize);
		return pageResult;
	}

	@RequestMapping("save")
	public Result save(@RequestBody TbBrand tbBrand) {

		Result result = new Result();

		if (tbBrand.getId() != null) {
			try {
				brandService.update(tbBrand);
				result.setSuccess(true);
				result.setMessage("存储成功");
			} catch (Exception e) {
				e.printStackTrace();
				result.setSuccess(false);
				result.setMessage("存储失败");
			}
			return result;
		}

		try {
			brandService.save(tbBrand);
			result.setSuccess(true);
			result.setMessage("存储成功");
		} catch (Exception e) {
			e.printStackTrace();
			result.setSuccess(false);
			result.setMessage("存储失败");
		}
		return result;
	}

	@RequestMapping("findOne")
	public TbBrand findOne(Long id) {
		TbBrand tbBrand = brandService.findOne(id);
		return tbBrand;
	}
	
	@RequestMapping("dele")
	public Result dele(Long[] ids) {
		Result result = new Result();
		try {
			brandService.delete(ids);
			result.setSuccess(true);
			result.setMessage("删除成功");
		} catch (Exception e) {
			e.printStackTrace();
			result.setSuccess(false);
			result.setMessage("删除失败");
		}
		return result;
	}
	
	
	@RequestMapping("/selectOptionList")
	public List<Map> selectOptionList(){
	return brandService.selectOptionList();
	}
	
	

}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller("brand", function ($scope, $controller, brandService) {
    //查询所有

    //继承
    $controller("baseController", { $scope: $scope })

    $scope.findAll = function () {
        brandService.findAll().success(function (response) {
            $scope.list = response;
        })
    };



    //分页查询	
    $scope.findPage = function (page, pageSize) {
        brandService.findPage(page, pageSize).success(function (response) {
            $scope.list = response.rows;
            $scope.paginationConf.totalItems = response.total;
        });
    }
    //存储
    $scope.save = function () {
        brandService.save($scope.entity).success(
            function (response) {
                if (response.flag) {
                    $scope.reloadList();
                } else {
                    alert(response.message);
                }
            })
    }

    $scope.findOne = function (id) {
        brandService.findOne(id).success(function (response) {
            $scope.entity = response;
        })
    }

    $scope.dele = function () {
        brandService.dele($scope.selectIds).success(function (response) {
            if (response.flag) {
                $scope.reloadList();
            } else {
                alter(response.message);
            }
        })

    }
    //搜索的方法
    $scope.tbBrand = {};
    $scope.search = function (page, pageSize) {
        brandService.search(page, pageSize, $scope.tbBrand).success(function (response) {
            $scope.list = response.rows;
            $scope.paginationConf.totalItems = response.total;
        })
    }

}); 